### PR TITLE
tar(StreamDirAsTar,StreamFileAsTar): fix possible fd leak and handle error from Stat()

### PR DESCRIPTION
**Pull request by php-coder**
PTAL @bparees

### CODE REVIEW

File: pkg/tar/tar.go [hunk 1]
```diff
@@ -91,17 +91,21 @@ func (t *stiTar) SetExclusionPattern(p *regexp.Regexp) {
...
 func (t *stiTar) StreamDirAsTar(source, dest string, writer io.Writer) error {
 	f, err := os.Open(source)
 	if err != nil {
 		return err
 	}
-	if info, _ := f.Stat(); !info.IsDir() {
+	defer f.Close()
+	info, err := f.Stat()
+	if err != nil {
+		return err
```
Comment (rhcarvalho) [inline pkg/tar/tar.go]: I don't know how this error gets exposed to the user, maybe it's possible to wrap it like below to include some more context.

Comment (php-coder) [inline pkg/tar/tar.go]: I handled it in the same way as we did for `os.Open()`. I believe that it's enough.

Comment (rhcarvalho) [inline pkg/tar/tar.go]: I think a useful context would be something like:

```go
return fmt.Errorf("streaming directory as tar: %v", err)
```

But depends on how this pops up to the end user, and if we want to be able to handle it from callers of `StreamDirAsTar`.

Comment (bparees) [inline pkg/tar/tar.go]: i think it's reasonable to make callers of StreamFileAsTar do the logging of the error if they so choose, which is the current behavior for errors on open as @php-coder noted anyway.

**Comment by bparees** [comment]
lgtm [merge]

**Comment by mfojtik** [comment]
I would prefer to holding off merging this if there is not significant bug associated with this till the rebase lands.

**Comment by bparees** [comment]
@mfojtik how does this affect the rebase?  this is in the s2i repo.

**Comment by mfojtik** [comment]
@bparees omg, ignore me... I thought this is a bump PR :-( oh my, I should get on PTO.

**Comment by php-coder** [comment]
[test] please

**Pull request merged**